sql/core/src/test/scala/org/apache/spark/sql/execution/UnsafeRowSerializerSuite.scala

```diff
@@ -25,6 +25,18 @@ import org.apache.spark.sql.catalyst.{CatalystTypeConverters, InternalRow}
 import org.apache.spark.sql.catalyst.expressions.{UnsafeProjection, UnsafeRow}
 import org.apache.spark.sql.types._
 
+
+/**
+ * used to test close InputStream in UnsafeRowSerializer
+ */
+class ClosableByteArrayInputStream(buf: Array[Byte]) extends ByteArrayInputStream(buf) {
```

Comment:
If you want to do this in a less-clear, more confusing way, you could use Mockito to `spy` a regular `ByteArrayInuptStream` then use Mockito assertions to check that `close()` was called. This is probably clearer, though, so no reason to resort to a complicated approach.